spec/helpers/request_helpers.rb

```diff
@@ -0,0 +1,57 @@
+module Helpers
+  module Controllers
+    def log_in
+      post '/users', params: {
+        user: attributes_for(:user)
+      }
+    end
+
+    def register_vehicle
+      log_in
+
+      post '/vehicles', params: {
+        vehicle: attributes_for(:toyota)
+      }
+    end
+
+    def create_ride_offer
+      log_in
+
+      register_vehicle
+
+      ride_attributes = attributes_for(:ride)
+      ride_attributes[:vehicle_id] = Vehicle.first.id
+      post '/rides', params: { ride: ride_attributes }
+    end
+
+    def register_different_user
+      user = create(:alexander)
+      post '/sessions', params: {
+        session: {
+          username_or_email: user.username,
+          login_password: 'Qwertyuiop123#'
+        }
+      }
+    end
+
+    def subscribe_to_ride_offer
```

Comment:
We had decided to do these things manually (i.e. without having to go through routing) right?

Reply:
We agreed on that only under the context that a user is logged out. Thus there would be no need to go through the routing. But a session is needed under the context that a user is logged in. Therefore, this helper method is necessary.